packages/state-importer/test/main.spec.js

```diff
@@ -0,0 +1,296 @@
+    })
+
+    it('should continue create on errors if `continueOnProblems`', async () => {
+      stateImport.continueOnProblems = true
+      stateImport._create.mockImplementationOnce(() =>
+        Promise.reject(new Error('First invalid state'))
+      )
+      stateImport._create.mockImplementationOnce(() =>
+        Promise.reject(new Error('Second invalid state'))
+      )
+      await stateImport._createOrUpdate(states, existingStates)
+      expect(stateImport._create).toHaveBeenCalledTimes(2)
+      expect(stateImport._summary.created).toBe(0)
+      expect(stateImport._summary.createErrorCount).toBe(2)
+      expect(stateImport._summary.errors).toHaveLength(2)
+      expect(stateImport._summary.errors[0]).toBe('First invalid state')
+      expect(stateImport._summary.errors[1]).toBe('Second invalid state')
+    })
+
+    it('should reject by default and stop on create error', async () => {
+      stateImport._create.mockImplementation(() =>
+        Promise.reject(new Error('Invalid new state'))
+      )
+      try {
+        await stateImport._createOrUpdate(states, existingStates)
+      } catch (error) {
+        // Put assertions in catch block because we expect promises to fail
+        expect(stateImport._create).toBeCalled()
+        expect(stateImport._summary.created).toBe(0)
+        expect(stateImport._summary.createErrorCount).toBe(2)
+        expect(stateImport._summary.errors).toHaveLength(2)
+        expect(stateImport._summary.errors[0]).toBe('Invalid new state')
+        expect(error).toEqual(new Error('Invalid new state'))
+      }
+    })
+  })
+
+  describe('::_update', () => {
+    const currentState = { ...states[1], id: 'some_Id' }
+
+    it('should be defined', () => {
+      expect(stateImport._update).toBeDefined()
+    })
+
+    it('should not call API if no update actions', async () => {
+      stateImport.client.execute = jest.fn(() => Promise.resolve())
+      const result = await stateImport._update(states[1], currentState)
+      expect(result).toEqual({ statusCode: 304 })
+      expect(stateImport.client.execute).not.toHaveBeenCalled()
+    })
+
+    it('should POST a state update', async () => {
+      stateImport.client.execute = jest.fn(() => Promise.resolve())
+      await stateImport._update(states[2], currentState)
+      expect(stateImport.client.execute).toHaveBeenCalled()
+    })
+  })
+
+  describe('::_create', () => {
+    it('should be defined', () => {
+      expect(stateImport._create).toBeDefined()
+    })
+
+    it('should POST a new state', async () => {
+      stateImport.client.execute = jest.fn(() => Promise.resolve())
+      await stateImport._create(states[0])
+      expect(stateImport.client.execute).toBeCalledWith(
+        expect.objectContaining({ body: states[0] })
+      )
+    })
+  })
+
+  describe('::_createService', () => {
+    it('should be defined', () => {
+      expect(stateImport._createService).toBeDefined()
+    })
+  })
```

Comment:
This is pretty subjective but I would structure tests to have setup in `beforeEach` and split cases into `describe`s.

Reply:
There's a beforeEach block:
```js
  beforeEach(() => {
    stateImport = new StateImport({ apiConfig }, logger)
  })
```
But as almost every test case requires a rather unique setup, I also do some of the setups in the test cases